registry/txt.go

```diff
@@ -243,6 +253,9 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 		if txt != nil {
 			txt.WithSetIdentifier(r.SetIdentifier)
 			txt.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
+			if im.isMigrationEnabled && r.Labels[endpoint.OwnerLabelKey] == im.oldOwnerID {
```

Comment:
This block is not tested. However, is not need to be tested, as legacy records deprecated https://github.com/kubernetes-sigs/external-dns/pull/5172